xbmc/dbwrappers/mysqldataset.cpp

```diff
@@ -147,7 +147,7 @@ int MysqlDatabase::connect(bool create_new) {
     return DB_CONNECTION_NONE;
 
   std::string resolvedHost;
-  if (CDNSNameCache::Lookup(host, resolvedHost))
+  if (!StringUtils::EqualsNoCase(host,"localhost") && CDNSNameCache::Lookup(host, resolvedHost))
```

Comment:
Is it only "localhost"? What about the usual local numeric IPv4 and IPv6 addresses, like 127.0.0.1 ?